Library/Homebrew/test/messages_spec.rb

```diff
@@ -2,49 +2,95 @@
+      messages.record_caveats(test_formula, "Zsh completions were installed")
+      expect(messages.caveats).to_not be_empty
+    end
   end
 
-  it "has recorded caveats" do
-    expect(@m.caveats).to_not be_empty
-  end
+  describe "#formula_installed" do
+    it "counts the number of formulas installed" do
+      expect(messages.formula_count).to eq 0
+      messages.formula_installed(test_formula, 1.1)
+      expect(messages.formula_count).to eq 1
+    end
 
-  it "maintained the order of recorded caveats" do
-    caveats_formula_order = @m.caveats.map { |x| x[:formula] }
-    expect(caveats_formula_order).to eq(["foo", "bar", "baz"])
+    it "adds to install_times" do
+      expect(messages.install_times).to be_empty
+      messages.formula_installed(test_formula, 1.1)
+      expect(messages.install_times).to_not be_empty
+    end
   end
 
-  it "has recorded installation times" do
-    expect(@m.install_times).to_not be_empty
+  describe "#display_messages" do
+    it "calls #display_caveats" do
+      expect(messages).to receive(:display_caveats)
+      messages.display_messages
+    end
+
+    context "when the --display-times argument is present" do
+      before { allow(ARGV).to receive(:include?).with("--display-times").and_return(true) }
+
+      it "calls #display_install_times" do
+        expect(messages).to receive(:display_install_times)
+        messages.display_messages
+      end
+    end
+
+    context "when the --display-times argument isn't present" do
+      before { allow(ARGV).to receive(:include?).with("--display-times").and_return(false) }
+
+      it "doesn't call #display_install_times" do
+        expect(messages).to_not receive(:display_install_times)
+        messages.display_messages
+      end
+    end
   end
 
-  it "maintained the order of install times" do
-    formula_order = @m.install_times.map { |x| x[:formula] }
-    expect(formula_order).to eq(["foo", "bar", "baz"])
+  describe "#display_caveats" do
+    context "when formula_count is less than two" do
+      it "doesn't print any output" do
+        messages.record_caveats(test_formula, "Zsh completions were installed")
+        messages.formula_installed(test_formula, 1.1)
+        expect { messages.display_caveats }.to_not output.to_stdout
+      end
+    end
+
+    context "when caveats is empty" do
+      it "doesn't print any output" do
+        messages.formula_installed(test_formula, 1.1)
+        expect { messages.display_caveats }.to_not output.to_stdout
+      end
+    end
+
+    context "when formula_count is greater than one and caveats are present" do
+      it "prints caveat details" do
+        test_formula2 = formula("foo") { url("https://example.com/foo-0.1.tgz") }
+        messages.record_caveats(test_formula, "Zsh completions were installed")
+        messages.formula_installed(test_formula, 1.1)
+        messages.formula_installed(test_formula2, 2.2)
+        expect { messages.display_caveats }.to output.to_stdout
```

Comment:
This should test what the output should look like. The same goes for all other tests which are currently only calling `.to output.to_stdout`.